Fix : search on rowid don't work with pgsql and natural search

atm-maxime opened the pull request:
Is this the right way ? Saying that any natural search should not search on rowid column ? or should we add a test in the natural_search function to use = instead of LIKE if we search on a "rowid" field ?

atm-maxime committed:
Fix : Product supplier price is related to supplier's currency

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Fix : search on rowid don't work with pgsql and natural search

eldy commented on htdocs/adherents/list.php:
```diff
@@ -83,7 +83,6 @@
 
 // List of fields to search into when doing a "search in all"
 $fieldstosearchall = array(
-    'd.rowid'=>'Ref',
```
I would prefer a test on 
if ($db->type == 'pgsql') to exclue the field only on posgresql
because search on ref is the common usage. Users won't understand why search on ref works everywhere, except here. If we limit this to postgresql user, we reduce this bad impact.

atm-maxime replied:
Agreed, I'll look into it.

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Merge branch '5.0' of https://github.com/Dolibarr/dolibarr into 5.0

atm-maxime committed:
Better fix only if bdd is pgsql

atm-maxime commented:
@eldy ok done

atm-maxime committed:
Merge branch '5.0' of github.com:atm-maxime/dolibarr into fix_member_search_pgsql

eldy committed:
Fix: with 5.0 and 6.0, supplier prices are always in main currency. The currency is not yet available in supplier prices.

Field to define the currency of supplier prices were added in database but not yet used by screen/code.
